RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractEdge.java

```diff
@@ -0,0 +1,197 @@
+
+public class ExtractEdge extends Extract{
+    
+    private static final Logger logger = Logger.getLogger(ExtractIE.class.getName());
+    private final IngestServices services = IngestServices.getInstance();
+    private final String moduleTempResultsDir;
+    private Content dataSource;
+    private IngestJobContext context;
+    
+    private static String ESE_TOOL_NAME = "ESEDatabaseView.exe";
+    private static File ESE_TOOL_FILE;
+    private static String EDGE_WEBCACHE_NAME = "WebCacheV01.dat";
+    private static String EDGE = "Edge";
+
+    ExtractEdge() throws NoCurrentCaseException{
+        moduleName = NbBundle.getMessage(Chrome.class, "ExtractEdge.moduleName");
+        moduleTempResultsDir = RAImageIngestModule.getRATempPath(Case.getCurrentCaseThrows(), EDGE) + File.separator + "results";
+    }
+    
+    @Override
+    void process(Content dataSource, IngestJobContext context) {
+        this.dataSource = dataSource;
+        this.context = context;
+        dataFound = false;
+        
+        this.processWebCache();
+        
+         // Bookmarks come from spartan.edb different file
+        this.getBookmark(); // Not implemented yet
+    }
+    
+    void processWebCache(){
+        Path path = Paths.get("ESEDatabaseView", ESE_TOOL_NAME);
+        ESE_TOOL_FILE =  InstalledFileLocator.getDefault().locate(path.toString(), ExtractEdge.class.getPackage().getName(), false); //NON-NLS
+        if (ESE_TOOL_FILE == null) {
+            this.addErrorMessage(
+                    NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.unableFindESEViewer", this.getName()));
+            logger.log(Level.SEVERE, "Error finding ESEDatabaseViewer program "); //NON-NLS
+        }
+        
+         final String esedumper = ESE_TOOL_FILE.getAbsolutePath();
+
+        // get WebCacheV01.dat files
+        org.sleuthkit.autopsy.casemodule.services.FileManager fileManager = currentCase.getServices().getFileManager();
+        List<AbstractFile> webCachFiles;
+        try {
+            webCachFiles = fileManager.findFiles(dataSource, EDGE_WEBCACHE_NAME); //NON-NLS
+        } catch (TskCoreException ex) {
+            this.addErrorMessage(NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.errGettingWebCacheFiles",
+                    this.getName()));
+            logger.log(Level.WARNING, "Error fetching 'index.data' files for Internet Explorer history."); //NON-NLS
+            return;
+        }
+
+        if (webCachFiles.isEmpty()) {
+            String msg = NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.noWebCachFiles");
+            logger.log(Level.INFO, msg);
+            return;
+        }
+
+        dataFound = true;   
+        
+        if(!PlatformUtil.isWindowsOS()){
+            logger.log(Level.WARNING, "Edge data found, unable to parse on non-windows system."); //NON-NLS
+            return;
+        }
+        
+        String temps;
+        String indexFileName;
```

Comment:
These local variables are only used in the for loop; it would be better in terms of "locality of reference" to define these inside of the loop. Note that there is no penalty for doing so - the compiler will do what it needs to do regarding space efficiency of local variables, so we need not concern ourselves with that and can follow the best practice of locality of reference.

Also, what is temps? I'm thinking tempFilePath or something similar would be better.

Reply:
That was lazy copy & paste.  Sorry about that.